src/test/utils/utils.ts

```diff
@@ -124,5 +124,49 @@ export async function validateRootNode(scan: any): Promise<[number, any]> {
   return [size, engines];
 }
 
+export async function waitForNotificationWithTimeout(timeout) {
+  let firstNotification;
+  let isTimeout = false;
+
+  const timer = setTimeout(() => {
+    isTimeout = true;
+  }, timeout);
+
+  while (!firstNotification) {
+    if (isTimeout) {
+      break;
+    }
+    const resultsNotifications = await new Workbench().getNotifications();
+    firstNotification = resultsNotifications[0];
+
+    await sleep(100);
```

Comment:
do we need the await command in the  method itself and also here?

Reply:
this is how Sara implemented it, Jusr refactored it to utils.ts in order to use it